Tests/ComposableArchitectureTests/StoreTests.swift

```diff
@@ -559,4 +559,169 @@ final class StoreTests: XCTestCase {
+      struct State: Equatable {
+        var count = 0
+      }
+      enum Action: Equatable {
+        case tap
+        case response1(Int)
+        case response2(Int)
+        case response3(Int)
+      } 
+      @Dependency(\.count) var count
+      func reduce(into state: inout State, action: Action) -> EffectTask<Action> {
+        switch action {
+        case .tap:
+          return withDependencies {
+            $0.count.value += 1
+          } operation: {
+            EffectTask.task { .response1(self.count.value) }
+          }
+          .eraseToEffect()
+        case let .response1(count):
+          state.count = count
+          return withDependencies {
+            $0.count.value += 1
+          } operation: {
+            EffectTask.task { .response2(self.count.value) }
+          }
+          .eraseToEffect()
+        case let .response2(count):
+          state.count = count
+          return withDependencies {
+            $0.count.value += 1
+          } operation: {
+            EffectTask.task { .response3(self.count.value) }
+          }
+          .eraseToEffect()
+        case let .response3(count):
+          state.count = count
+          return .none
+        }
+      }
+    }
+
+    let testStore = TestStore(
+      initialState: Feature.State(),
+      reducer: Feature()
+    )
+    await testStore.send(.tap)
+    await testStore.receive(.response1(1)) {
+      $0.count = 1
+    }
+    await testStore.receive(.response2(1))
+    await testStore.receive(.response3(1))
+
+    let store = Store(
+      initialState: Feature.State(),
+      reducer: Feature()
+    )
+    await store.send(.tap)?.value
+    XCTAssertEqual(store.state.value.count, testStore.state.count)
+  }
+}
+
+private struct Count: TestDependencyKey {
+  var value: Int
+  static let liveValue = Count(value: 0)
+  static let testValue = Count(value: 0)
+}
+extension DependencyValues {
+  fileprivate var count: Count {
+    get { self[Count.self] }
+    set { self[Count.self] = newValue }
+  }
+}
+
+
+public struct _Observe<Reducers: ReducerProtocol>: ReducerProtocol {
+  @usableFromInline
+  let reducers: (State, Action) -> Reducers
+
+  /// Initializes a reducer that builds a reducer from the current state and action.
+  ///
+  /// - Parameter build: A reducer builder that has access to the current state and action.
+  @inlinable
+  public init(
+    @ReducerBuilder<Reducers.State, Reducers.Action> _ build: @escaping (Reducers.State, Reducers.Action) -> Reducers
+  ) where Reducers.State == State, Reducers.Action == Action {
+    self.init(internal: build)
+  }
+
+  @usableFromInline
+  init(
+    internal reducers: @escaping (Reducers.State, Reducers.Action) -> Reducers
+  ) {
+    self.reducers = reducers
+  }
+
+  @inlinable
+  public func reduce(
+    into state: inout Reducers.State, action: Reducers.Action
+  ) -> EffectTask<Reducers.Action> {
+    self.reducers(state, action).reduce(into: &state, action: action)
+  }
 }
```

Comment:
It may need a few adjustments to build on Swift 5.8.
```suggestion
public struct _Observe<State, Action, Reducers: ReducerProtocol>: ReducerProtocol
where Reducers.State == State, Reducers.Action == Action {
  @usableFromInline
  let reducers: (State, Action) -> Reducers

  /// Initializes a reducer that builds a reducer from the current state and action.
  ///
  /// - Parameter build: A reducer builder that has access to the current state and action.
  @inlinable
  public init(
    @ReducerBuilder<State, Action> _ build: @escaping (State, Action) -> Reducers
  ) {
    self.init(internal: build)
  }

  @usableFromInline
  init(
    internal reducers: @escaping (State, Action) -> Reducers
  ) {
    self.reducers = reducers
  }

  @inlinable
  public func reduce(
    into state: inout Reducers.State, action: Reducers.Action
  ) -> EffectTask<Reducers.Action> {
    self.reducers(state, action).reduce(into: &state, action: action)
  }
}
```

Reply:
ha sorry, that was accidentally committed from messing with due to that slack conversation. we are not shipping that type in this PR.